File: mpi/mpi_Initialized.m
function [flag, info, toplevel] = mpi_Initialized
%mpi_Initialized	Indicates whether mpi_Init has been called
%
%  [flag, info] = mpi_Initialized
%
%  flag(int) 0 false
%	         1 true
%
%  info(int) return code
%
%  SEE ALSO: mpi_Init, mpi_Finalize, mpi_Finalized
%
% C interface:
% int MPI_Initialized(int *flag )
% http://mpi.deino.net/mpi_functions/MPI_Initialized.html

%#codegen -args {}

coder.cinclude('mpi.h');

flag = int32(0); info = int32(0); %#ok<NASGU>

info = coder.ceval('MPI_Initialized', coder.wref(flag));

toplevel = nargout>2;
if info && (toplevel || m2c_debug)
    m2c_error('MPI:RuntimeError', 'MPI_Initialized returned error code %d\n', info)
end

File: mpi/codegen/lib/mpi_Initialized/mpi_Initialized.c
#include "mpi_Initialized.h"
#include "mpi.h"
#include "petscsys.h"
#include "petscksp.h"
#include "m2c.h"

static void m2c_error(int varargin_3);
static void m2c_error(int varargin_3)
{
  M2C_error("MPI:RuntimeError", "MPI_Initialized returned error code %d\n",
            varargin_3);
}

void mpi_Initialized(int *flag, int *info, boolean_T *toplevel)
{
  *info = MPI_Initialized(flag);
  *toplevel = true;
  if (*info != 0) {
    m2c_error(*info);
  }
}

void mpi_Initialized_initialize(void)
{
}

void mpi_Initialized_terminate(void)
{
}
